src/pamiq_core/torch/trainer.py

```diff
@@ -0,0 +1,224 @@
+"""PyTorch trainer implementation for pamiq-core.
+
+This module provides a base class for implementing PyTorch model
+training within the pamiq-core framework. It handles optimizer and
+learning rate scheduler configuration, state management, and integrates
+with the pamiq-core training system.
+"""
+
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, cast, override
+
+import torch
+import torch.nn as nn
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import LRScheduler
+
+from pamiq_core.torch import TorchTrainingModel
+from pamiq_core.trainer import Trainer
+
+# Type definitions for improved type safety and readability
+type StateDict = dict[str, Any]
+type OptimizersDict = dict[str, Optimizer]
+type LRSchedulersDict = dict[str, LRScheduler]
+type OptimizersSetup = OptimizersDict | tuple[OptimizersDict, LRSchedulersDict]
+
+
+class TorchTrainer(Trainer):
+    """Base class for PyTorch model training in pamiq-core.
+
+    This trainer integrates PyTorch models with the pamiq-core framework,
+    providing functionality for optimizer configuration, state persistence,
+    and model type validation. It automatically handles the setup and teardown
+    of optimizers and learning rate schedulers during the training process.
+
+    Subclasses should implement the `configure_optimizers` and `train` methods
+    to define the specific training behavior.
+    """
+
+    def __init__(
+        self,
+        training_condition_data_user: str | None = None,
+        min_buffer_size: int = 0,
+        min_new_data_count: int = 0,
+    ) -> None:
+        """Initialize the PyTorch trainer.
+
+        Sets up empty containers for optimizers, schedulers, and their
+        respective states. Actual optimizer and scheduler instances will
+        be created during the setup phase.
+
+        Args:
+            training_condition_data_user: Name of the data user to check for trainability.
+                If None, trainer is always trainable.
+            min_buffer_size: Minimum total data points required in the buffer.
+            min_new_data_count: Minimum number of new data points required since last training.
+        """
+        super().__init__(
+            training_condition_data_user,
+            min_buffer_size,
+            min_new_data_count,
+        )
+
+        # Containers for optimizer and scheduler instances
+        self.optimizers: OptimizersDict = {}
+        self.lr_schedulers: LRSchedulersDict = {}
```

Comment:
`self.schedulers`としてしまった方が`self.optimizers`と文字数が揃うからより見通しをよくできる気がする
```suggestion
        self.schedulers: LRSchedulersDict = {}
```

Reply:
`pamiq-core`ではシステムのイベント処理をスケジューリングするためにも`Scheduler`という言葉を用いているので、その重複を避けるために`lr_`をつけている

Reply:
名前の重複があるなら確かに話は別になるかも　`lr_`あった方がよさそう